x-pack/plugins/endpoint/server/routes/metadata/query_builders.ts

```diff
@@ -46,37 +49,43 @@ export const kibanaRequestToMetadataListESQuery = async (
 };
 
 async function getPagingProperties(
-  request: KibanaRequest<any, any, any>,
+  request: KibanaRequest<unknown, unknown, MetadataIndexGetBodyResult>,
   endpointAppContext: EndpointAppContext
 ) {
   const config = await endpointAppContext.config();
   const pagingProperties: { page_size?: number; page_index?: number } = {};
-  if (request?.body?.paging_properties) {
+  if (request.body?.paging_properties) {
     for (const property of request.body.paging_properties) {
-      Object.assign(
-        pagingProperties,
-        ...Object.keys(property).map(key => ({ [key]: property[key] }))
-      );
+      if ('page_size' in property) {
+        pagingProperties.page_size = property.page_size;
+      } else {
+        pagingProperties.page_index = property.page_index;
+      }
     }
   }
+  // If page_size or page_index are 0 or undefined, use the defaults instead. TODO, is this logic right?
```

Comment:
TODO, can someone confirm that this behavior is correct?

Reply:
Also the comment is kinda wrong. The value will be a number or null. If the number is 0 or null it'll be falsey and get the default.